src/gadgets/curves/bn256/ec_add.rs

```diff
@@ -0,0 +1,18 @@
+use crate::{cs::traits::cs::ConstraintSystem, gadgets::curves::SmallField};
+
+use super::*;
+
+/// Adds two points in the plain SW projective coordinates.
+pub fn projective_add<F, CS>(
+    cs: &mut CS,
+    point_1: &mut BN256SWProjectivePoint<F>,
+    point_2: BN256SWProjectivePoint<F>,
```

Comment:
since we already get X and Y coordinates passed through in the ecadd call, it seems superfluous to cast point 2 to projective and then clone the points out. we should just pass a tuple of coords